src/firebolt/async_db/connection.py

```diff
@@ -45,6 +47,15 @@
 logger = logging.getLogger(__name__)
 
 
+def _validate_engine_name_and_url(
```

Comment:
The reason I split them was in case we want to actually validate the engine with a request, and since requests would have to be sync/async there's a duplication. However, I think I'm overengineering this, as we're changing this functionality for Identity anyway. I'll consolidate.